src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts

```diff
@@ -0,0 +1,221 @@
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import {
+  TreeComponent,
+  ITreeOptions,
+  TreeModel,
+  TreeNode,
+  TREE_ACTIONS
+} from '@circlon/angular-tree-component';
+import { NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { Subscription, Observable } from 'rxjs';
+import { RgwDaemonService } from '~/app/shared/api/rgw-daemon.service';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { ModalService } from '~/app/shared/services/modal.service';
+import { TimerService } from '~/app/shared/services/timer.service';
+import { RgwMultisiteFormComponent } from '../rgw-multisite-form/rgw-multisite-form.component';
+
+@Component({
+  selector: 'cd-rgw-multisite-details',
+  templateUrl: './rgw-multisite-details.component.html',
+  styleUrls: ['./rgw-multisite-details.component.scss']
+})
+export class RgwMultisiteDetailsComponent implements OnDestroy, OnInit {
+  private sub = new Subscription();
+
+  @ViewChild('tree') tree: TreeComponent;
+
+  icons = Icons;
+  permission: Permission;
+  selection = new CdTableSelection();
+  createTableActions: CdTableAction[];
+  editTableActions: CdTableAction[];
+
+  loadingIndicator = true;
+  nodes: any[] = [];
+  treeOptions: ITreeOptions = {
+    useVirtualScroll: true,
+    nodeHeight: 22,
+    levelPadding: 20,
+    actionMapping: {
+      mouse: {
+        click: this.onNodeSelected.bind(this)
+      }
+    }
+  };
+
+  realms: any = [];
+  zonegroups: any = [];
```

Comment:
can we create interfaces for better type checking? I would strongly suggest to not use `any` whenever we can.

Reply:
Yeah, that was on my to-do list, will update